Userland/Libraries/LibWeb/IndexedDB/IDBFactory.cpp

```diff
@@ -25,4 +33,64 @@ void IDBFactory::initialize(JS::Realm& realm)
     WEB_SET_PROTOTYPE_FOR_INTERFACE(IDBFactory);
 }
 
+// https://w3c.github.io/IndexedDB/#dom-idbfactory-open
+JS::ThrowCompletionOr<JS::NonnullGCPtr<IDBOpenDBRequest>> IDBFactory::open(String const& name, Optional<u64> version)
+{
+    auto& realm = this->realm();
+
+    // 1. If version is 0 (zero), throw a TypeError.
+    if (version.has_value() && version.value() == 0)
+        return realm.vm().throw_completion<JS::TypeError>(JS::ErrorType::InvalidVersion, 0);
+
+    // 2. Let environment be this's relevant settings object.
+    auto& environment = HTML::relevant_settings_object(*this);
+
+    // 3. Let storageKey be the result of running obtain a storage key given environment.
+    //    If failure is returned, then throw a "SecurityError" DOMException and abort these steps.
+    auto storage_key = StorageAPI::obtain_a_storage_key(environment);
+    if (!storage_key.has_value())
+        return realm.vm().throw_completion<WebIDL::SecurityError>("Failed to obtain a storage key"_string);
+
+    // 4. Let request be a new open request.
+    auto request = IDBOpenDBRequest::create(realm);
+
+    // 5. Run these steps in parallel:
+    Platform::EventLoopPlugin::the().deferred_invoke(JS::create_heap_function(realm.heap(), [&realm, storage_key, name, version, request] {
+        HTML::TemporaryExecutionContext context(realm, HTML::TemporaryExecutionContext::CallbacksEnabled::Yes);
+
+        // 1. Let result be the result of opening a database connection, with storageKey, name, version if given and undefined otherwise, and request.
+        auto result = open_a_database_connection(realm, storage_key.value(), name, version, request);
+
+        // 2. Queue a task to run these steps:
+        HTML::queue_a_task(HTML::Task::Source::DatabaseAccess, nullptr, nullptr, JS::create_heap_function(realm.heap(), [&realm, &request, result]() {
```

Comment:
Mark this lambda mutable for it to allow you to modify result. should probably also do a `result = move(result)`.

so,
```
HTML::queue_a_task(HTML::Task::Source::DatabaseAccess, nullptr, nullptr, JS::create_heap_function(realm.heap(), [&realm, &request, result = move(result)]() mutable {
```